components/service/glean/src/main/java/mozilla/components/service/glean/storages/StorageEngineManager.kt

```diff
@@ -14,9 +15,17 @@ import org.json.JSONObject
 internal class StorageEngineManager(
     private val storageEngines: Map<String, StorageEngine> = mapOf(
         "events" to EventsStorageEngine,
-        "strings" to StringsStorageEngine
-    )
+        "strings" to StringsStorageEngine,
+        "uuids" to UuidsStorageEngine
+    ),
+    applicationContext: Context
```

Comment:
Maybe unrelated to this PR: `StorageEngineManager` seems to be responsible to keep the map of storage engines. Do the engines themselves need to be singletons then? Or can they get accessed through the storage engine manager? This would simplify this setup here too; where you need to pass the applicationContext to the singletons.

Reply:
Good point. The downside of this would be that `StorageEngineManager` would then need to be a singleton (the *specific metric APIs* require either this or the engines to be one). Moreover, when calling the engine from the API (e.g. `UuidMetricType`), we'd need to manually upcast from the `StorageEngine` returned by the manager (each engine may implements custom methods not available on the interface, which are used by the APIs).

We might revisit this post MVP though.

Reply:
`StorageEngineManager` sounds like a Singleton tbh :)

Reply:
Fair enough! Let me file a bug to deal with that (we're tracking our current tech debt on bugzilla :) ).  I filed bug 1491345 for this!